main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"flag"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/Wattpad/kube-sqs-autoscaler/scale"
+	"github.com/Wattpad/kube-sqs-autoscaler/sqs"
+)
+
+var (
+	pollInterval        = flag.Duration("poll-period", 5*time.Second, "The interval in seconds for checking if scaling is required")
+	scaleDownCoolPeriod = flag.Duration("scale-down-cool-down", 30*time.Second, "The cool down period for scaling down")
+	scaleUpCoolPeriod   = flag.Duration("scale-up-cool-down", 10*time.Second, "The cool down period for scaling up")
+	scaleUpMessages     = flag.Int("scale-up-messages", 100, "Number of sqs messages queued up required for scaling up")
+	scaleDownMessages   = flag.Int("scale-down-messages", 10, "Number of messages required to scale down")
+	maxPods             = flag.Int("max-pods", 5, "Max pods that kube-sqs-autoscaler can scale")
+	minPods             = flag.Int("max-pods", 1, "Min pods that kube-sqs-autoscaler can scale")
+	awsRegion           = flag.String("aws-region", "us-east-1", "Your AWS region")
+
+	sqsQueueUrl              = flag.String("sqs-queue-url", "", "The sqs queue url")
+	kubernetesDeploymentName = flag.String("kubernetes-deployment", "", "Kubernetes Deployment to scale. This field is required")
+)
+
+func Run(p *scale.PodAutoScaler, sqs *sqs.SqsClient) {
```

Comment:
hmm I was thinking we can do e2e testing using minikube like we did for nginx-ingress-controller but now that I think about it you're probably right we can probably mock that too. Let me take a stab at that